### PR TITLE
Allow toggling samples & regions in add things step UI #1378

### LINKED ISSUE #1305: Update project collection workflow shows child physical things inconsistently
I'm not certain of the exact intended behavior, but something looks inconsistent in this workflow.

1. Add a phys thing to a project
2. Take a bunch of samples
3. Update project collection: compare number of physical things to physical things in sidebar, make changes, save, reopen

https://github.com/archesproject/arches-for-science/assets/38668450/9db77ce2-c8d6-4472-840e-29251c0c350f

### LINKED ISSUE #1378: Add things step UI improvements for child physical things
- [x] Add search toggles to include/exclude samples, analysis areas (affecting left side only)
- [x] Sort the right side alphabetically (by parent)
- [x] Group children physical things under parents
- [x] Fix wrapping of names

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Closes #1378

- Add toggles for filtering samples/analysis areas out of search results
- Sort collection in sidebar alphabetically by parent
- Group and indent children under parent

### Demo ###
Screenshot 2023-10-26 at 12 02 01 PM

### CODE REVIEW

File: arches_for_science/media/js/views/components/workflows/sample-taking-workflow/sample-taking-sample-location-step.js [hunk 1]
```diff
@@ -343,7 +343,7 @@ define([
                 }
             }
 
-            const tilesBelongingToManifest = this.tilesBelongingToManifest(self.card.tiles(), self.canvases());
+            const tilesBelongingToManifest = this.tilesBelongingToManifest(self.card.tiles());
```
Comment (jacobtylerwalls) [inline arches_for_science/media/js/views/components/workflows/sample-taking-workflow/sample-taking-sample-location-step.js:346]: Unrelated cleanup re #1304; this function no longer takes two args, missed it before merge.

File: arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js [hunk 2]
```diff
@@ -273,8 +334,11 @@ define([
                     self.termOptions = results;
 
                     const filteredResults = results.filter(function(result){
-                        return result.context_label.includes("Physical Thing") ||
-                        result.context_label.includes("Search Term");
+                        return (
+                            result.context_label.includes("Physical Thing") ||
+                            result.context_label.includes("Search Term")
+                        ) && (self.includeSamples() || !result.text.includes(sampleSubstring))
+                        && (self.includeAnalysisAreas() || !result.text.includes(regionSubstring))
```
Comment (jacobtylerwalls) [inline arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js:341]: I still found substring matching necessary to filter out children from the select2 widget, as I don't have access to concepts/tiles at that point. Open to other ideas.

**Comment by jacobtylerwalls** [comment]
@chiatt ready for a look!

File: arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js (hunk 2 above)
Comment (chiatt) [reply arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js:341]: Maybe we could do something like this on the Arches side: https://github.com/archesproject/arches/pull/10216/files

Comment (jacobtylerwalls) [reply arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js:341]: I don't think the graphid will help, because both samples and their parents will share the same graph id for physical thing. AFAICT I would need to drill in to the tile data along the lines of what I did in the middle of the long sort function added in this PR.

Comment (jacobtylerwalls) [reply arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js:341]: Just tested it out, and this is all I get back with archesproject/arches#10216:
```json
        {
            "context": "",
            "context_label": "Physical Thing - Name",
            "graphid": "9519cb4f-b25b-11e9-8c7b-a4d18cec433a",
            "id": 1,
            "text": "qwe [Sample of Alabastron (2003.192)]",
            "type": "term",
            "value": "qwe [Sample of Alabastron (2003.192)]"
        },
        {
            "context": "",
            "context_label": "Physical Thing - Name",
            "graphid": "9519cb4f-b25b-11e9-8c7b-a4d18cec433a",
            "id": 2,
            "text": "qwe [Sample Area of Alabastron (2003.192)]",
            "type": "term",
            "value": "qwe [Sample Area of Alabastron (2003.192)]"
        }
```

**Review by chiatt — APPROVED** [review]
This looks great. I think we can address the issue with filtering the dropdown results once this work is completed: https://github.com/archesproject/arches-for-science/issues/1435

File: arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js (hunk 2 above)
Comment (chiatt) [reply arches_for_science/media/js/views/components/workflows/create-project-workflow/add-things-step.js:341]: I see - right, it's the distinction between types of physical things that is needed, not resource models.
